Adds the (fast) mold linker

Hofer-Julian opened the pull request:
In order to use it in a flatpak:

1. "org.freedesktop.Sdk.Extension.llvm12" has to be added next to this extension (for clang)
2. `.cargo/config.toml` with the following content has to be addedd

```toml
[target.x86_64-unknown-linux-gnu]
linker = "clang"
rustflags = ["-C", "link-arg=-fuse-ld=/usr/lib/sdk/rust-stable/bin/mold"]
```

bilelmoussaoui commented:
I think it would be great to document the existence of mold in the README along with the config to make use of it.

Hofer-Julian commented:
> I think it would be great to document the existence of mold in the README along with the config to make use of it.

Sounds good.

I will go to sleep now, but if you want feel free to push to this branch :)

Hofer-Julian committed:
Adds the (fast) mold linker

In order to use it in a flatpak:

1. Add "org.freedesktop.Sdk.Extension.llvm12" next to this extension in order to get clang
2. Add `.cargo/config.toml` with the following content

```toml
[target.x86_64-unknown-linux-gnu]
linker = "clang"
rustflags = ["-C", "link-arg=-fuse-ld=/usr/lib/sdk/rust-stable/bin/mold"]
```

Hofer-Julian committed:
Add gitignore

Hofer-Julian committed:
Add README and mention mold

Hofer-Julian committed:
mold: Update url

bilelmoussaoui commented on README.md:
```diff
@@ -0,0 +1,17 @@
+
+In order to use the (fast) [mold](https://github.com/rui314/mold) linker:
+
+1. Add "org.freedesktop.Sdk.Extension.llvm12" next to this extension in order to get clang
```
I would mention that is only needed until gcc12 lands on the freedesktop runtime. Let me ask around when that can happen

bilelmoussaoui replied:
Seems there is no tarballs for gcc12 yet, so I would mention the llvm12 is needed until there is a release of gcc12. then we will have it it in fdo 22.08 once it is out

Hofer-Julian replied:
Done

Erick555 replied:
I would recommend using llvm13 here which is latest version on flathub

Hofer-Julian committed:
Mold: mention that clang is only necessary until gcc12.1 release

Erick555 commented on README.md:
```diff
@@ -0,0 +1,20 @@
+
+In order to use the (fast) [mold](https://github.com/rui314/mold) linker:
+
+1. Add "org.freedesktop.Sdk.Extension.llvm12" next to this extension in order to get clang
```
I would recommend using llvm13 here which is latest version on flathub

Hofer-Julian replied:
Done

Hofer-Julian committed:
Recommend llvm 13 instead

alatiera commented:
Looks good :+1:
